.github/workflows/e2e.yml

```diff
@@ -54,14 +57,123 @@ jobs:
           retention-days: 30
 
       - name: Build Wallet
-        if: ${{ needs.diff.outputs.isWallet == 'true' }}
+        # need to run Wallet e2e when its upstream(TS SDK and Rust) or itself is changed
+        if: ${{ needs.diff.outputs.isWallet == 'true' || needs.diff.outputs.isRust == 'true' || needs.diff.outputs.isTypescriptSDK == 'true'}}
         run: pnpm wallet build
       - name: Run Wallet e2e tests
-        if: ${{ needs.diff.outputs.isWallet == 'true' }}
+        if: ${{ needs.diff.outputs.isWallet == 'true' || needs.diff.outputs.isRust == 'true' || needs.diff.outputs.isTypescriptSDK == 'true'}}
         run: xvfb-run --auto-servernum --server-args="-screen 0 1280x960x24" -- pnpm wallet playwright test
       - uses: actions/upload-artifact@v3
         if: always()
         with:
           name: playwright-report-wallet
           path: apps/wallet/playwright-report/
           retention-days: 30
+
+  # Run e2e test against localnet built on the devnet branch for backward compatibility check
+  local_devnet_branch:
+    name: Local Network Built on devnet branch
+    needs: diff
+    # TODO: add wallet e2e to the `if` condition when available
+    if: needs.diff.outputs.isExplorer == 'true' || needs.diff.outputs.isTypescriptSDK == 'true'
```

Comment:
We should be able to get away with just the SDK tests right? Ideally they're comprehensive enough to validate backwards compatibility unless I'm missing something?

Reply:
Not 100% - suppose the return type for `ExecuteTransaction` changes from `TransactionResponseV1` to `TransactionResponseV2`, and the TS-SDK now supports both. People might be just tempted to update the Explorer to use `TransactionResponseV2` and this will break backward compatibility if the explorer tests are skipped

Reply:
Makes sense, thanks for the clarification!